homeassistant/components/zha/binary_sensor.py

```diff
@@ -60,258 +70,103 @@
-            )
-
-    async def async_added_to_hass(self):
-        """Run when about to be added to hass."""
-        await super().async_added_to_hass()
-        old_state = await self.async_get_last_state()
-        if self._state is not None or old_state is None:
-            return
-
-        _LOGGER.debug("%s restoring old state: %s", self.entity_id, old_state)
-        if 'level' in old_state.attributes:
-            self._level = old_state.attributes['level']
-        self._state = old_state.state == STATE_ON
-
-    async def async_update(self):
-        """Retrieve latest state."""
-        from zigpy.zcl.clusters.general import OnOff
-        result = await helpers.safe_read(
-            self._endpoint.out_clusters[OnOff.cluster_id],
-            ['on_off'],
-            allow_cache=False,
-            only_cache=(not self._initialized)
-        )
-        self._state = result.get('on_off', self._state)
-
-
-class BinarySensor(RestoreEntity, ZhaEntity, BinarySensorDevice):
-    """ZHA switch."""
-
-    _domain = DOMAIN
-    _device_class = None
-    value_attribute = 0
-
-    def __init__(self, device_class, **kwargs):
-        """Initialize the ZHA binary sensor."""
-        super().__init__(**kwargs)
-        self._device_class = device_class
-        self._cluster = list(kwargs['in_clusters'].values())[0]
-
-    def attribute_updated(self, attribute, value):
-        """Handle attribute update from device."""
-        _LOGGER.debug("Attribute updated: %s %s %s", self, attribute, value)
-        if attribute == self.value_attribute:
-            self._state = bool(value)
-            self.async_schedule_update_ha_state()
-
-    async def async_added_to_hass(self):
-        """Run when about to be added to hass."""
-        await super().async_added_to_hass()
-        old_state = await self.async_get_last_state()
-        if self._state is not None or old_state is None:
-            return
-
-        _LOGGER.debug("%s restoring old state: %s", self.entity_id, old_state)
-        self._state = old_state.state == STATE_ON
-
-    @property
-    def cluster(self):
-        """Zigbee cluster for this entity."""
-        return self._cluster
-
-    @property
-    def zcl_reporting_config(self):
-        """ZHA reporting configuration."""
-        return {self.cluster: {self.value_attribute: REPORT_CONFIG_IMMEDIATE}}
-
     @property
-    def is_on(self) -> bool:
-        """Return if the switch is on based on the statemachine."""
-        if self._state is None:
-            return False
-        return self._state
-
-    @property
-    def device_class(self) -> str:
-        """Return device class from component DEVICE_CLASSES."""
-        return self._device_class
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        if LEVEL in self._temporary_state_attributes:
+            level = self._temporary_state_attributes.get(LEVEL, 0)
```

Comment:
We don't need to both check and use `dict.get`. Use one of those approaches.

Reply:
0 is a valid value when a level is actually present. This is like this because only remote device binary sensors have a level attribute. Doing this like this ensures I only show this attribute when it is actually an attribute that should be shown.

Reply:
gone